### PR TITLE
feat: Async handlers use person info from event

### PR DESCRIPTION
**Pull request by tiina303**
## Problem

Whenever we process webhooks we would query the person information from Postgres this is not great for 2 reasons:
1. Unneccesary load to Postgres
2. The person might have changed already, e.g. imagine a user becomes paying user and tied to paying user actions webhooks are sent, but the events before they actually became a paying user shouldn't be sent to the webhook for paying users only.

## Changes

Instead of querying Postgres we're simply using the person info we already have on the event - that's tied to the person at the time of the event.

Note that for some reason the person_created_at value is saved as CH second precision ts (if we mess the conversion up in async handlers, then we get `null` as the ts in webhooks handling) - made sure we cover this in tests and added a comment too.

⚠️  I'm removing for what we send to REST (e.g. Zapier) webhooks:
-  `person.id` , if we wanted to keep that we'd need to query PG, that doesn't seem reasonable.
- `person.team_id` - that's obviously always the same as the event teamId, no reason to send it again?

^ discussion: https://posthog.slack.com/archives/C01MGUHFH6G/p1682966993974429
✅  Checked via `select team_id, * from ee_hook` it's only Zapier in US and EU and that's confirmed in the thread that it's ok.

## How did you test this code?

Ran PostHog locally:
- Added an `ee_hook` row with `action_performed = 'action_performed'`, `resource_id` set to the action's IDs (set it up for the `test-event` in localhost:8000), and target pointing to webhook.site
  - `insert into ee_hook VALUES (current_timestamp, current_timestamp, 'action_performed', 'https://webhook.site/faac4a48-af48-4b38-8480-dc1a09a6131e', 1, 1, 1, 1);`
  - made zapier available
- sent the test event 
  - https://webhook.site/#!/faac4a48-af48-4b38-8480-dc1a09a6131e/88d1f427-9c4b-44cc-9c5d-16bc40770166/1

### CODE REVIEW

File: plugin-server/src/utils/db/db.ts [hunk 1]
```diff
@@ -1134,17 +1134,18 @@ export class DB {
...
     }
 
-    public async doesPersonBelongToCohort(cohortId: number, person: IngestionPersonData): Promise<boolean> {
+    public async doesPersonBelongToCohort(cohortId: number, personUuid: string, teamId: number): Promise<boolean> {
         const psqlResult = await this.postgresQuery(
             `
             SELECT count(1) AS count
             FROM posthog_cohortpeople
             JOIN posthog_cohort ON (posthog_cohort.id = posthog_cohortpeople.cohort_id)
+            JOIN (SELECT * FROM posthog_person where team_id = $3) AS posthog_person_in_team ON (posthog_cohortpeople.person_id = posthog_person_in_team.id)
```
Comment (tiina303) [inline plugin-server/src/utils/db/db.ts:1143]: checked it works in metabase http://metabase-eu/question#eyJkYXRhc2V0X3F1ZXJ5Ijp7InR5cGUiOiJuYXRpdmUiLCJuYXRpdmUiOnsicXVlcnkiOiItLSBTRUxFQ1QgY291bnQoMSkgQVMgY291bnRcbnNlbGVjdCAqXG4gICAgICAgICAgICBGUk9NIHBvc3Rob2dfY29ob3J0cGVvcGxlXG4gICAgICAgICAgICBKT0lOIHBvc3Rob2dfY29ob3J0IE9OIChwb3N0aG9nX2NvaG9ydC5pZCA9IHBvc3Rob2dfY29ob3J0cGVvcGxlLmNvaG9ydF9pZClcbiAgICAgICAgICAgIEpPSU4gKFNFTEVDVCAqIEZST00gcG9zdGhvZ19wZXJzb24gd2hlcmUgdGVhbV9pZCA9IDMwMDIpIEFTIHBvc3Rob2dfcGVyc29uX2luX3RlYW0gT04gKHBvc3Rob2dfY29ob3J0cGVvcGxlLnBlcnNvbl9pZCA9IHBvc3Rob2dfcGVyc29uX2luX3RlYW0uaWQpXG4gICAgICAgICAgICBXSEVSRSBjb2hvcnRfaWQ9MTI2NVxuICAgICAgICAgICAgLS0gICBBTkQgcG9zdGhvZ19wZXJzb25faW5fdGVhbS5pZD03MzcyNzY0MFxuICAgICAgICAgICAgICBBTkQgcG9zdGhvZ19wZXJzb25faW5fdGVhbS51dWlkPScwMTg2YWVhNS0zYzMxLTAwMDAtNGFjOS03ODY3YjVmYjhmODknXG4gICAgICAgICAgICAtLSAgIEFORCB0ZWFtX2lkPTMwMDJcbiAgICAgICAgICAgIC0tICAgQU5EIHBlcnNvbl9pZD03MzcyNzY0MFxuICAgICAgICAgICAgICBBTkQgcG9zdGhvZ19jb2hvcnRwZW9wbGUudmVyc2lvbiBJUyBOT1QgRElTVElOQ1QgRlJPTSBwb3N0aG9nX2NvaG9ydC52ZXJzaW9uXG5saW1pdCA1IiwidGVtcGxhdGUtdGFncyI6e319LCJkYXRhYmFzZSI6MzR9LCJkaXNwbGF5IjoidGFibGUiLCJwYXJhbWV0ZXJzIjpbXSwidmlzdWFsaXphdGlvbl9zZXR0aW5ncyI6e319

File: plugin-server/src/worker/ingestion/hooks.ts [hunk 2]
```diff
@@ -255,23 +242,19 @@ export class HookCommander {
...
+            const properties = event.person_properties
+            const team_id = event.teamId
+            const createdAt = event.person_created_at
 
             sendablePerson = {
+                // TODO: can we drop the id from here?
                 uuid,
                 properties,
                 team_id,
-                id,
```
Comment (tiina303) [inline plugin-server/src/worker/ingestion/hooks.ts:274]: Note: we're dropping the person.id from being sent with webhooks - if we don't want to drop that we'll need to do a postgres query - I suspect its better for us to drop it, users shouldn't know the PG person id

**Comment by tiina303** [comment]
unrelated known flake is failing for functional tests - that was also fixed on master (just haven't rebased)

File: plugin-server/tests/worker/ingestion/action-matcher.test.ts [hunk 3]
```diff
@@ -692,35 +678,23 @@ describe('ActionMatcher', () => {
...
+            // ])
+            // expect(await actionMatcher.match({...event, person_properties: {foo: 'baR'}})).toEqual([])
+            // expect(await actionMatcher.match({...event, person_properties: {foo: 'baz'}})).toEqual([])
+            // expect(await actionMatcher.match({...event, person_properties: {foo: 'barabara'}})).toEqual([])
+            // expect(await actionMatcher.match({...event, person_properties: {foo: 'rabarbar'}})).toEqual([])
+            // expect(await actionMatcher.match({...event, person_properties: {foo: 7}})).toEqual([])
+            // expect(await actionMatcher.match({...event, person_properties: {}})).toEqual([])
+            // expect(await actionMatcher.match({...event, person_properties: {something_else: 999}})).toEqual([])
+            // expect(await actionMatcher.match({...event, person_properties: {foo: true}})).toEqual([])
+            // expect(await actionMatcher.match({...event, person_properties: {foo: null}})).toEqual([])
```
Comment (Twixes) [inline plugin-server/tests/worker/ingestion/action-matcher.test.ts:697]: Why's this commented out? This shouldn't be committed

**Review by Twixes — COMMENTED** [review]
Looks good, just found some commented-out code

File: plugin-server/tests/worker/ingestion/action-matcher.test.ts (hunk 3 above)
Comment (tiina303) [reply plugin-server/tests/worker/ingestion/action-matcher.test.ts:697]: ah ... I was testing something and forgot ... good spot

**Review by hazzadous — APPROVED** [review]

**Pull request merged**